polaris/polaris/management/commands/poll_pending_deposits.py

```diff
@@ -135,9 +141,45 @@ def execute_deposits(cls):
                 "poll_pending_deposits() returned None. "
                 "Ensure is returns a list of transaction objects."
             )
+        distribution_accounts = {}
+        # Add the transactions that the anchor has been collecting signatures for
+        # but are now ready to submit.
+        # TODO: PR REVIEWERS, should we ask the anchor which transactions should
+        #  be added to the list of ready transactions or try to query them ourselves
+        #  like so?
+        #  We should definitely provide an integration function or explict
+        #  Transaction boolean column if the query below could in theory collect
+        #  transactions that are not ready. From my understanding, all Transaction
+        #  objects collected with the attribute values below _would_ be ready.
+        #  The idea is that anchors change the _status_ and _pending_signatures_
+        #  field to signal to Polaris that the object is ready.
+        ready_transactions.extend(
+            list(
+                Transaction.objects.filter(
+                    kind=Transaction.KIND.deposit,
+                    status=Transaction.STATUS.pending_anchor,
+                    pending_signatures=False,
+                    envelope__isnull=False,
+                )
+            )
+        )
         for transaction in ready_transactions:
             if module.TERMINATE:
                 break
+            if transaction.pending_signatures:
+                transaction.status = Transaction.STATUS.pending_anchor
+                asset_code = transaction.asset.code
+                if asset_code not in distribution_accounts:
+                    distribution_accounts[
+                        asset_code
+                    ] = settings.HORIZON_SERVER.load_account(
+                        transaction.asset.distribution_account
+                    )
+                transaction.envelope = create_transaction_envelope(
+                    transaction, distribution_accounts[asset_code]
```

Comment:
@JakeUrban why not pass `asset.distribution_account`? why is there a need for this `distribution_accounts` dict?

Reply:
`asset.distribution_account` is a public key string. `settings.HORIZON_SERVER.load_account()` returns an `stellar_sdk.account.Account` object, which is whats required by `create_transaction_envelope()`.

Currently we're caching previously requested `Account` objects for the respective `distribution_account`s. But I'm realizing we need to refresh the accounts to get updated sequence numbers.

Reply:
Actually, the sequence number will likely be the same if the transaction envelope created in the last iteration has not been submitted. 

This is an issue because Polaris never attempts to maintain an internal sequence number for an account, it always just pulls it from Horizon. But since we're not submitting these transactions right away, the sequence numbers will be wrong by the time they are submitted.

This means Polaris needs to maintain an internal sequence number for every anchored asset's distribution account.

Reply:
This also means that ideally, Polaris is the only service that submits transactions from the asset's distribution accounts. If this isn't the case, Polaris will have to request the correct sequence number from Horizon and update existing envelopes with sequence numbers _less than the correct sequence number_.

Reply:
Turns out updating sequence numbers would invalidate signatures, which makes sense. The solution is to submit the transaction from a Stellar account _derived_ from the distribution account. 

If every multisig transaction has uses a derived source account, we can:
1. Fund the account
2. Make the payment using that account
3. Merge the account back to the distribution account

This ensures no mulisig transaction will be rejected for sequence number-related reasons. The downside is added performance cost when submitting these transactions.

Reply:
@JakeUrban If I understood correctly, you're suggesting that for each transaction:
1. create a new account derived from `distribution_account`
2. use the new account as the channel source and initially sign the envelope with it
3. let the anchor add the signatures to the envelope
4. submit the transaction
5. merge the new accoun back to the distribution account

That works, since it removes the need to worry about the sequence numbers and also allows multiple Polaris instances using the same distribution account.
Channels seem very useful, maybe it would be interesting to have channel support in Polaris in the future for batch transaction submitting.